modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/IgniteProject.java

```diff
@@ -191,6 +196,27 @@ public IgniteProject(RelInput input) {
         return ImmutableList.of(Pair.of(nodeTraits.replace(collation), ImmutableList.of(in)));
     }
 
+    /** */
+    @Override public List<Pair<RelTraitSet, List<RelTraitSet>>> passThroughCorrelation(RelTraitSet nodeTraits,
+        List<RelTraitSet> inTraits) {
+        if (RexUtils.hasCorrelation(getProjects()))
+            return ImmutableList.of();
```

Comment:
The empty list should be returned only in case when correlations derived from `getProjects()` is not subset of `TraitUtils.correlation(nodeTraits))`

Reply:
Fixed